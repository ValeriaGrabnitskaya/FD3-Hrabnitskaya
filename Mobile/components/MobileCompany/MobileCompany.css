.MobileCompany
{
    font-family: "Lucida Sans Unicode", "Lucida Grande", Sans-Serif;
    font-size: 14px;
    border-collapse: collapse;
    text-align: center;
    margin: 0 auto;
    margin-bottom: 20px;
}

.MobileCompany .MobileCompanyFilterButtons
{
    margin: auto;
    margin-bottom: 10px;
}

.MobileCompany .MobileCompanyButton 
{
    margin-right: 10px;
    padding: 7px 10px 7px 10px;
    background-color: rgb(50, 100, 300);
    color: white;
    font-size: 14px;
    border: 0;
}

.MobileCompany .MobileCompanyButton:hover
{
    background-color: rgb(0, 0, 179);
    color: aliceblue;
}

.MobileCompany .MobileCompanyClients
{
    margin: auto;
    margin-bottom: 10px;
    border-collapse: collapse;
}

.MobileCompany .MobileCompanyClients .TableHeader
{
    border-bottom: 1px solid black;
    height: 50px;
}