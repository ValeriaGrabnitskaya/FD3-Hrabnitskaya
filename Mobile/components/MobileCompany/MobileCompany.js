import React from 'react';
import PropTypes from 'prop-types';

import './MobileCompany.css';
import MobileClient from './MobileClient/MobileClient';
import clientStatuses from '../../static-data/client-statuses';
import AddEditClient from '../AddEditClient/AddEditClient';
import modes from '../../static-data/mode';
import { companyEvents } from '../events';

class MobileCompany extends React.PureComponent {

    static propTypes = {
        clients: PropTypes.arrayOf(
            PropTypes.shape({
                id: PropTypes.number.isRequired,
                firstName: PropTypes.string.isRequired,
                secondName: PropTypes.string.isRequired,
                lastName: PropTypes.string.isRequired,
                balance: PropTypes.number.isRequired,
                status: PropTypes.string.isRequired
            })
        )
    };

    state = {
        clients: this.props.clients,
        mode: null,
        selectedClientId: null
    };

    componentDidMount = () => {
        companyEvents.addListener('SaveClient', this.saveClient);
        companyEvents.addListener('EditClient', this.editClient);
        companyEvents.addListener('CancelSaveClient', this.cancelSaveClient);
        companyEvents.addListener('DeleteClient', this.deleteClient);
    };

    componentWillUnmount = () => {
        companyEvents.removeListener('SaveClient', this.saveClient);
        companyEvents.removeListener('EditClient', this.editClient);
        companyEvents.removeListener('CancelSaveClient', this.cancelSaveClient);
        companyEvents.removeListener('DeleteClient', this.deleteClient);
    };

    saveClient = (savedClient) => {
        if (this.state.mode === modes.add) {
            this.setState({ clients: [...this.state.clients, savedClient], mode: modes.none });
        }
        if (this.state.mode === modes.edit) {
            let copiedClients = [...this.state.clients];
            copiedClients.forEach((client, i) => {
                if (client.id == savedClient.id) {
                    copiedClients.splice(i, 1, savedClient)
                }
            });
            this.setState({ clients: copiedClients, mode: modes.none });
        }
    }

    editClient = (selectedId) => {
        this.setState({ mode: modes.edit, selectedClientId: selectedId });
    }

    cancelSaveClient = () => {
        this.setState({ mode: modes.none, editClient: null })
    }

    deleteClient = (selectedId) => {
        let copiedClients = [...this.state.clients];
        copiedClients.forEach((client, i) => {
            if (client.id == selectedId) {
                copiedClients.splice(i, 1);
            }
        });
        this.setState({ clients: copiedClients })
    }

    getFilteredClients = (status) => {
        switch (status) {
            case clientStatuses.Active:
                this.filterClients(clientStatuses.Active);
                break;
            case clientStatuses.Blocked:
                this.filterClients(clientStatuses.Blocked);
                break;
            default:
                this.filterClients();
                break;
        }
    }

    filterClients = (status) => {
        var filteredArray = [];
        if (status) {
            filteredArray = this.props.clients.filter((client) => {
                return client.status === status
            })
        } else {
            filteredArray = this.props.clients;
        }

        this.setState({ clients: filteredArray })
    }

    addClient = () => {
        this.setState({ mode: modes.add, editClient: null })
    }

    render() {
        console.log("MobileCompany render");

        var clients = this.state.clients.map(client =>
            <MobileClient key={client.id} client={client} />
        );

        var selectedClient = this.state.clients.find((client) => client.id === this.state.selectedClientId);

        var isAddMode = this.state.mode == modes.add;
        var isEditMode = this.state.mode == modes.edit;

        return (
            <div className='MobileCompany'>
                <input type="button" value="Все" onClick={() => this.getFilteredClients()} />
                <input type="button" value="Активные" onClick={() => this.getFilteredClients(clientStatuses.Active)} />
                <input type="button" value="Заблокированные" onClick={() => this.getFilteredClients(clientStatuses.Blocked)} />
                <table className='MobileCompanyClients'>
                    <thead>
                        <tr>
                            <th>Фамилия</th>
                            <th>Имя</th>
                            <th>Отчество</th>
                            <th>Баланс</th>
                            <th>Статус</th>
                            <th>Редактировать</th>
                            <th>Удалить</th>
                        </tr>
                    </thead>
                    <tbody>{clients}</tbody>
                </table>
                <input type="button" value="Добавить клиента" onClick={this.addClient} />
                {
                    (isAddMode || isEditMode) && <AddEditClient mode={this.state.mode} client={selectedClient} />
                }
            </div>
        )
    }
}

export default MobileCompany