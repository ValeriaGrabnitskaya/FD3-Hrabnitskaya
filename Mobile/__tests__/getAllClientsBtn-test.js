"use strict";

import renderer from 'react-test-renderer';
import React from 'react';

import MobileCompany from '../components/MobileCompany/MobileCompany';

var clients = require('../data/clients.json');

test('проверка фильтрации всех клиентов', () => {

    const MobileCompanyComponent = renderer.create(
        <MobileCompany clients={clients}/>
    );

    let componentTree = MobileCompanyComponent.toJSON();
    expect(componentTree).toMatchSnapshot();

    var activeFilterBtn = MobileCompanyComponent.root.find((el) => el.props.id == 'allClientsBtn')
    activeFilterBtn.props.onClick();

    componentTree=MobileCompanyComponent.toJSON();
    expect(componentTree).toMatchSnapshot();
});