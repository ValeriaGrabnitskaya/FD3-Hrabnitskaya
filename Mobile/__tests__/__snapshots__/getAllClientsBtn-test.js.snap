// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка фильтрации всех клиентов 1`] = `
<div
  className="MobileCompany"
>
  <div
    className="MobileCompanyFilterButtons"
  >
    <input
      className="MobileCompanyButton"
      id="allClientsBtn"
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      className="MobileCompanyButton"
      id="activeFilterBtn"
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      className="MobileCompanyButton"
      id="blockedFilterBtn"
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <table
    className="MobileCompanyClients"
  >
    <thead>
      <tr>
        <th
          className="TableHeader"
        >
          Фамилия
        </th>
        <th
          className="TableHeader"
        >
          Имя
        </th>
        <th
          className="TableHeader"
        >
          Отчество
        </th>
        <th
          className="TableHeader"
        >
          Баланс
        </th>
        <th
          className="TableHeader"
        >
          Статус
        </th>
        <th
          className="TableHeader"
        >
          Редактировать
        </th>
        <th
          className="TableHeader"
        >
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="ClientAttribute"
        >
          Иванов
        </td>
        <td
          className="ClientAttribute"
        >
          Иван
        </td>
        <td
          className="ClientAttribute"
        >
          Иванович
        </td>
        <td
          className="ClientAttribute"
        >
          100
        </td>
        <td
          className="ClientAttribute ActiveStatus"
        >
          active
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="editBtn"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="deleteBtn"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td
          className="ClientAttribute"
        >
          Петров
        </td>
        <td
          className="ClientAttribute"
        >
          Пётр
        </td>
        <td
          className="ClientAttribute"
        >
          Петрович
        </td>
        <td
          className="ClientAttribute"
        >
          200
        </td>
        <td
          className="ClientAttribute ActiveStatus"
        >
          active
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="editBtn"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="deleteBtn"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td
          className="ClientAttribute"
        >
          Сидоров
        </td>
        <td
          className="ClientAttribute"
        >
          Сидр
        </td>
        <td
          className="ClientAttribute"
        >
          Сидорович
        </td>
        <td
          className="ClientAttribute"
        >
          300
        </td>
        <td
          className="ClientAttribute ActiveStatus"
        >
          active
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="editBtn"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="deleteBtn"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td
          className="ClientAttribute"
        >
          Григорьев
        </td>
        <td
          className="ClientAttribute"
        >
          Григорий
        </td>
        <td
          className="ClientAttribute"
        >
          Григорьевич
        </td>
        <td
          className="ClientAttribute"
        >
          -50
        </td>
        <td
          className="ClientAttribute BlockedStatus"
        >
          blocked
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="editBtn"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="deleteBtn"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    className="MobileCompanyButton"
    id="addBtn"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`проверка фильтрации всех клиентов 2`] = `
<div
  className="MobileCompany"
>
  <div
    className="MobileCompanyFilterButtons"
  >
    <input
      className="MobileCompanyButton"
      id="allClientsBtn"
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      className="MobileCompanyButton"
      id="activeFilterBtn"
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      className="MobileCompanyButton"
      id="blockedFilterBtn"
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <table
    className="MobileCompanyClients"
  >
    <thead>
      <tr>
        <th
          className="TableHeader"
        >
          Фамилия
        </th>
        <th
          className="TableHeader"
        >
          Имя
        </th>
        <th
          className="TableHeader"
        >
          Отчество
        </th>
        <th
          className="TableHeader"
        >
          Баланс
        </th>
        <th
          className="TableHeader"
        >
          Статус
        </th>
        <th
          className="TableHeader"
        >
          Редактировать
        </th>
        <th
          className="TableHeader"
        >
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="ClientAttribute"
        >
          Иванов
        </td>
        <td
          className="ClientAttribute"
        >
          Иван
        </td>
        <td
          className="ClientAttribute"
        >
          Иванович
        </td>
        <td
          className="ClientAttribute"
        >
          100
        </td>
        <td
          className="ClientAttribute ActiveStatus"
        >
          active
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="editBtn"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="deleteBtn"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td
          className="ClientAttribute"
        >
          Петров
        </td>
        <td
          className="ClientAttribute"
        >
          Пётр
        </td>
        <td
          className="ClientAttribute"
        >
          Петрович
        </td>
        <td
          className="ClientAttribute"
        >
          200
        </td>
        <td
          className="ClientAttribute ActiveStatus"
        >
          active
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="editBtn"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="deleteBtn"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td
          className="ClientAttribute"
        >
          Сидоров
        </td>
        <td
          className="ClientAttribute"
        >
          Сидр
        </td>
        <td
          className="ClientAttribute"
        >
          Сидорович
        </td>
        <td
          className="ClientAttribute"
        >
          300
        </td>
        <td
          className="ClientAttribute ActiveStatus"
        >
          active
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="editBtn"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="deleteBtn"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td
          className="ClientAttribute"
        >
          Григорьев
        </td>
        <td
          className="ClientAttribute"
        >
          Григорий
        </td>
        <td
          className="ClientAttribute"
        >
          Григорьевич
        </td>
        <td
          className="ClientAttribute"
        >
          -50
        </td>
        <td
          className="ClientAttribute BlockedStatus"
        >
          blocked
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="editBtn"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="ClientAttribute"
        >
          <input
            className="Input"
            id="deleteBtn"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    className="MobileCompanyButton"
    id="addBtn"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;
